File: app/src/main/java/com/blog/demo/control/picker/NumberPickerActivity.java
package com.blog.demo.control.picker;

import android.app.Activity;
import android.os.Bundle;
import android.widget.NumberPicker;

import androidx.annotation.Nullable;

import com.blog.demo.LogTool;
import com.blog.demo.R;

public class NumberPickerActivity extends Activity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_control_picker_number);

        NumberPicker numberPicker = findViewById(R.id.number_picker);
        numberPicker.setMinValue(10);
        numberPicker.setMaxValue(30);
        numberPicker.setValue(20);
        numberPicker.setOnValueChangedListener(new NumberPicker.OnValueChangeListener() {
            @Override
            public void onValueChange(NumberPicker picker, int oldVal, int newVal) {
                LogTool.logi("NumberPickerActivity", oldVal + ":" + newVal);
            }
        });
    }
}

File: app/src/main/java/com/blog/demo/control/picker/NumberPickerActivity.kt
package com.blog.demo.control.picker

import android.app.Activity
import android.os.Bundle
import android.widget.NumberPicker
import com.blog.demo.LogTool
import com.blog.demo.R

class NumberPickerActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_control_picker_number)

        val numberPicker: NumberPicker = findViewById(R.id.number_picker)
        numberPicker.minValue = 10
        numberPicker.maxValue = 30
        numberPicker.value = 20
        numberPicker.setOnValueChangedListener { _, oldVal, newVal ->
            LogTool.logi("NumberPickerActivity", "$oldVal:$newVal")}
    }

}